@ws docs (#339)
architect/arc.codes: src/views/docs/en/reference/app.arc/ws.md

---
app: testapp

ws: ~
```

</div>
</arc-tab>

</div>
</arc-viewer>

Running `arc create` generates the following functions:

```bash
/
├── ws
│   ├── connect
│   ├── default
│   ├── disconnect
├── app.arc
└── package.json
```

> Each function responds to websocket events from a client.

Reviewer: ```suggestion
Each function responds to WebSocket events from a client.
```

> In the functions payload there is a connectionId that uniquely identifies a client. This connectionId can be used to send messages to the correct client.

Reviewer: ```suggestion
In the [payload delivered to the function](#function-payload) there is a `connectionId` that uniquely identifies a client. This `connectionId` can be used to send messages to the correct client.
```

### Functions

> The functions created by the @ws pragma handle events from a websocket client.

Reviewer: ```suggestion
The functions created by the `@ws` pragma handle events from a WebSocket client.
```

> * Connect - This function is invoked when a websocket client connects to the application

Reviewer: ```suggestion
* Connect - This function is invoked when a WebSocket client connects to the application
```

> * Default - This function is invoked when a websocket client sends a message to the application

Reviewer: ```suggestion
* Default - This function is invoked when a WebSocket client sends a message to the application
```

> * Disconnect - This function is invoked when a websocket client disconnects from your application

Reviewer: ```suggestion
* Disconnect - This function is invoked when a WebSocket client disconnects from your application
```

#### Function Payload

|Argument|Description|
|---|---|

> |`req`|The websocket request payload|

Reviewer: ```suggestion
|`req`|The WebSocket request payload|
```

|`req.requestContext.connectionId`|An id that uniquely identifies a client|

#### Send Messages

> To publish a message to a websocket client you can use the architect/functions ws.send method. You can call this method from any of your applications functions.

Reviewer: ```suggestion
To publish a message to a WebSocket client you can use arc's runtime library `@architect/functions`' `ws.send` method. You can call this method from any of your application's functions.
```
